src/Umbraco.Core/Routing/UrlProvider.cs

```diff
@@ -24,7 +26,7 @@ public class UrlProvider : IPublishedUrlProvider
         /// <param name="mediaUrlProviders">The list of media url providers.</param>
         /// <param name="variationContextAccessor">The current variation accessor.</param>
         /// <param name="propertyEditorCollection"></param>
-        public UrlProvider(IUmbracoContextAccessor umbracoContextAccessor, IWebRoutingSettings routingSettings, UrlProviderCollection urlProviders, MediaUrlProviderCollection mediaUrlProviders, IVariationContextAccessor variationContextAccessor)
+        public UrlProvider(IUmbracoContextAccessor umbracoContextAccessor, IOptionsSnapshot<WebRoutingSettings> routingSettings, UrlProviderCollection urlProviders, MediaUrlProviderCollection mediaUrlProviders, IVariationContextAccessor variationContextAccessor)
         {
             if (routingSettings == null) throw new ArgumentNullException(nameof(routingSettings));
```

Comment:
From what I read [here](https://tpodolak.com/blog/2018/01/29/asp-net-core-make-sure-ioptionst-initialized/), and in testing it out, I'm not sure we actually need a guard clause here at all any more.  It seems even if we've omitted to configure this particular class (in `AddUmbracoConfiguration`), due to the fact we're injecting in via `IOptions`, it'll never be null (if unconfigured, it would be non-null, with default values).  So have just removed this check.

Reply:
Makes sense! 🙂